compiler/noirc_evaluator/src/ssa/opt/inlining.rs

```diff
@@ -122,9 +122,13 @@ impl Ssa {
         let mut new_functions = std::collections::BTreeMap::new();
         for entry_point in inline_targets {
             let function = &self.functions[&entry_point];
-            let inlined = function.inlined(&self, &should_inline_call)?;
+            let inlined = function.inlined(&self, &should_inline_call, &mut callees)?;
             new_functions.insert(entry_point, inlined);
         }
+        for callee in callees {
+            // If we have callees that were not inlined, we need to keep them in the SSA.
```

Comment:
It feels unideal that we have to start threading a callers set just so that we can not remove incorrectly marked ACIR functions called from Brillig. Could we looks at avoiding this `callers` set? 

We could try marking ACIR called from Brillig as an entry point within the inline infos or we just error out during inlining if we ever have an ACIR function called from Brillig. We already have Brillig functions we do not inline that remain in place as they are marked as should not inline by the inline infos and they do not require this tracking. Could we use a similar strategy and just mark ACIR called from Brillig as should not inline thus triggering them as an inline target? The other option would be to just error during inlining which I think is reasonable as it is a bad formed SSA.

Reply:
yes, in fact this is not enough in some more complex cases and I did not manage to make it through without inlining.
so I will remove this and simply return an error.

Reply:
Done